File: legacy/src/logic/solvers/EulerSolver.java
package logic.solvers;

import model.physics.Body2D;
import model.physics.Universe2D;

public 
class EulerSolver
extends Physics2DSolver
{
	public 
	EulerSolver(Universe2D u) 
	{
		super(u);
		timestep = 0.01;
		displayName = "Euler";
	}
	
	@Override
	public Physics2DSolver getNewInstance(Universe2D u) {
		return new EulerSolver(u);
	}

	@Override public 
	void step() 
	{
		computeAllBodiesAccelerations();
		
		//velocity & position loop
		for(Body2D b1 : universe.bodies)
		{
			b1.position.add(b1.velocity.times(timestep));
			b1.velocity.add(b1.acceleration.times(timestep));
		}
		
		timeElapsed += timestep;
	}
}

File: src/physics/solvers/euler_solver.cpp
/*
 * euler_solver.cpp
 *
 *  Created on: 12 de jul de 2016
 *      Author: carlosfaruolo
 */

#include "euler_solver.hpp"

#include <vector>
using std::vector;

const AbstractPhysics2DSolver::CustomFactory<EulerSolver>
EulerSolver::CLASS_FACTORY("EulerSolver", "Euler");

EulerSolver::EulerSolver(Universe2D& u)
: AbstractPhysics2DSolver(&CLASS_FACTORY, u, 0.01)
{}

void EulerSolver::step()
{
	computeAllBodiesAccelerations();

	//velocity & position loop
	foreach(Body2D*, b1p, vector<Body2D*>, universe.bodies)
	{
		Body2D& b1 = *b1p;
		b1.position += b1.velocity * timestep;
		b1.velocity += b1.acceleration * timestep;
	}

	timeElapsed += timestep;
}


